packages/kit/src/core/prerender/prerender.js

```diff
@@ -401,7 +401,14 @@ export async function prerender() {
 				validate_common_exports(page.shared, route.id);
 			}
 
-			const prerender = get_option(nodes, 'prerender') ?? false;
+			const should_prerender = get_option(nodes, 'prerender');
+			const prerender =
+				// Try prerendering if ssr is false and no server needed.
+				// People can opt out of this behavior by explicitly setting prerender to false
+				(should_prerender !== false &&
+					get_option(nodes, 'ssr') === false &&
+					!page?.server?.actions) ||
+				(should_prerender ?? false);
```

Comment:
mhhhhm I originally had `nodes.some(node => node.server?.load` as a condition in there, but removed it because that is  not called and prerendered - BUT, I think the server manifest will not include this route then anymore. Is my hunch correct? What can we do about that besides "don't prerender if there's any server load function"? Is this also true for shared load functions? Need to test this ..

Reply:
confirmed, it doesn't work for nodes that have any server load functions, because they are left out of the server manifest, which results in a 404.

Reply:
fixed through setting it to `'auto'` in this case